### ISSUE TITLE
Stuck on 1st wallet, never logs out, even after clearing playerprefs

### ISSUE THREAD

**Issue by LiamVisionary** [ChainSafe/web3.unity#140]
**Describe the bug**
Hi, I've been having this issue for weeks now and no matter what I do, I can't figure out whats wrong. When someone logs into my project with their first account, the web3 connect scene never prompts them to login again and it always connects to the first account ever logged into on the game. I've set up a number of methods to clear the wallet or disconnect but even none of these give me any results:

```
        _walletLogOut.OnLogOut();
        PlayerPrefs.DeleteAll();
        PlayerPrefs.SetString("Account", "");
```

under 

```
    async private void OnConnected()
    {
        _walletLogOut.OnLogOut();
        PlayerPrefs.DeleteAll();
        PlayerPrefs.SetString("Account", "");
        Debug.Log(PlayerPrefs.GetString("Account") + " found.");
        account = ConnectAccount();
        while (account == "") {
            await new WaitForSeconds(1f);
            account = ConnectAccount();
        };
        // save account for next scene
        PlayerPrefs.SetString("Account", account);
        // reset login message
        SetConnectAccount("");
        // load next scene
        SceneManager.LoadScene(SceneManager.GetActiveScene().buildIndex + 1);
    }

```

The oddest thing.. is that everything seems to work fine through Unity's localhost build from hitting "Build and Run". But the moment I host it on Itch.io, I get the issue with the wallet being stuck on the first ever logged in.

Also even stranger is that, even after disconnecting the wallet in metamask, it still is able to connect in the login scene without asking to sign in -- and it connects to that first account. Really baffled here..

**To Reproduce**
1. 'Build for WebGL
2. Compress to Zip and upload to Itch.io
3. Embed game on website.
4. Press connect on the login scene.

**Expected behavior**
To connect with the most recently connected account.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: MacOS Mojave
 - Browser Chrome
 - Version 100.0.4

**Comment by KBryan** [comment]
Try using the OnSignOut prefab and see if that fixes your issue. The code is pretty basic.
`
 // Clear Account
PlayerPrefs.SetString("Account", "0x0000000000000000000000000000000000000001");
// go to login scene
SceneManager.LoadScene(0);
`

**Comment by LiamVisionary** [comment]
Thanks for the response @KBryan, I've already added the Web GL Sign Out script to a logout button on my title screen, and while it runs the above code to nullify the account, I suspect the issue isn't with the PlayerPrefs, but rather with the SDK not attempting to sign in again on the connect scene and defaulting to the first account. Also keep in mind this seems to only under one of the three scenarios/combination thereof:

1) game is compressed to .zip
2) game is uploaded to itch.io
3) game embedded through itch in html

I know also that itch has their own compression algorithm. I'm not too familiar with that or with the inner workings of the chainsafe sdk, but could the problem lie therein?

**Comment by LiamVisionary** [comment]
Is it possible that the iframe embed has a scope of its own, and that even though I disconnect my wallet on my website, it's remaining logged in on the iframe? Asking since, the issue is nonexistent when running locally through unity's build and run, which I suspect is not using an iframe.

**Comment by KBryan** [comment]
It seems like this is an issue with itch.io. Seems like your local environment doesn't have an issue.

**Comment by LiamVisionary** [comment]
@KBryan Suppose I've gotta find an alternative hosting solution. Is there any chance of this being looked into with Itch.io, seeing as how popular itch is as a WebGL game hosting platform?

**Comment by LiamVisionary** [comment]
@KBryan After plenty of debugging, I've found that the issue is actually in using iframes to embed the unity player. I switched over to AWS S3 and metamask interfaces as it should using their direct link: https://fantom-survivor.s3.amazonaws.com/index.html
but when I embed using an <iframe>, it stops interfacing with metamask after the first account login, and theres no way of disconnecting or switching from that account. Seems that the ChainSafe Web3 API is incompatible with iframes. Might be a worthwhile addition in the docs.
